projects/memapCore/src/main/java/memap/helper/milp/MILPSolver.java

```diff
@@ -229,80 +331,60 @@ private void workWithResults(double[] optSolution, String[] names, double[] lamb
 			for (int i = 0; i < currentDemand.length -1; i++) {
 				resultWithTotalHeatDemand[0] += currentDemand[i];
 				resultWithTotalHeatDemand[i+1] = currentDemand[i];
-			}			
+			}
 			resultWithTotalHeatDemand[currentDemand.length] = currentDemand[currentDemand.length-1];			
 			currentDemand = resultWithTotalHeatDemand;
-		}
-
-		String[] namesResult = HelperConcat.concatAllObjects(timeStep, currentDemandNames, currentOptVectorNames,
-				currentSOCNames, currentStorageEnergyContentNames, energyPricesNames, totalCosts, co2emissions);
-		double[] vectorResult = HelperConcat.concatAlldoubles(currentStep, currentDemand, currentOptVector, currentSOC,
-				currentStorageEnergyContent, currentEnergyPrices, totalCostsEUR, totalCO2emissions);
-
+		}		
+
+		String[] namesResult_this_TimeStep = HelperConcat.concatAllObjects(
+				timeStep, 
+				currentDemandNames, 
+				currentOptVectorNames, 				
+				currentEfficiencyNames,				
+				currentSOCNames, 
+				currentStorageEnergyContentNames, 
+				energyPricesNames, 
+				totalCosts, 
+				co2emissions);
 		
+		double[] vectorResult_this_TimeStep = HelperConcat.concatAlldoubles(
+				currentStep, 
+				currentDemand, 
+				currentOptVector,				
+				currentEfficiencyValues,				
+				currentSOC,
+				currentStorageEnergyContent, 
+				currentEnergyPrices, 
+				totalCostsEUR, 
+				totalCO2emissions);
+
+		// FIXME - check if the "ToolUsage.SERVER" part has to be adapted to the changes above as well.
```

Comment:
I think this is not needed. Only ConnectionDB.addResults() in ln 368 has to be changed a bit according to that new values.